### PR TITLE
AP-621 (b) non liquid cap, pensioner disregard and populate results in Disposable Capital Assessment

### PR DESCRIPTION
**Pull request by stephenrichards**
## Finalise Disposable Capital Asssessment service

[Link to story](https://dsdmoj.atlassian.net/browse/AP-621)

- Changed DatedStruct to output JSON without the intermediary `table:` key that is output by its parent `OpenStruct`
- Added the `NonLiquidCapitalAssessment` service
- Added the pensioner disregard values to the thresholds config files
- Added the `PensionerCapitalDisregard` service
- Added code to the `DisposableCapitalAssessment` service to add up the populate the results and thresholds in the response structure

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/dated_struct.rb [hunk 1]
```diff
@@ -13,6 +17,17 @@ def []=(name, value)
...
 
+  # OpenStruct will normally serialize to JSON with an intermediate key 'table'.
+  # DatedStruct will only do it if created with option {serialize_as_open_struct: true}
+  def to_hash
+    if @options[:serialize_as_open_struct]
+      instance_values.except!('options').as_json
+    else
+      to_h
+    end
+    # @options[:serialize_as_open_struct] ? super : to_h
```
Comment (mpw5) [inline app/models/dated_struct.rb]: Should this comment be here?

File: app/services/workflow_service/pensioner_capital_disregard.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      raise 'No disposable income specified for non-passported applicant' if disposable_income.nil?
+
+      disregard_value(disposable_income)
+    end
+
+    private
+
+    def pensioner?
+      earliest_dob_for_pensioner
+      @submission_date - minimum_pensioner_age.years >= applicant_dob
```
Comment (mpw5) [inline app/services/workflow_service/pensioner_capital_disregard.rb]: Should these two lines just be `earliest_dob_for_pensioner >= applicant_dob`?

File: config/thresholds/8-Apr-2019.yml [hunk 3]
```diff
@@ -9,6 +9,21 @@ disposable_income_lower: 100
 disposable_income_upper: 50000
 capital_lower: 1
 capital_upper: 30000
+pensioner_capital_disregard:
+  passported: 100_000
+  # disregard to use if monthly disposable income (excluding income derived from capital) is eqqual to or below these figures
```
Comment (mpw5) [inline config/thresholds/8-Apr-2019.yml]: Very nitpicky - there should be only one `q` in `equal`

File: public/schemas/assessment_response.json [hunk 4]
```diff
@@ -72,6 +75,10 @@
           "description": "Aggregate value of lowest balances of bank accounts during the calculation period",
           "$ref": "http://localhost:3000/schemas/assessment_request.json#definitions/positive_number_with_two_decimals"
         },
+        "non_liquid_capital_assessment": {
+          "description": "Aggregate value of all non-liquid capital itesm excluding property and vehicles, e.g. stocks and shares, valuable items, jewellery",
```
Comment (mpw5) [inline public/schemas/assessment_response.json]: Nitpicky again - `itesm` should be `items`

File: app/models/dated_struct.rb (hunk 1 above)
Comment (stephenrichards) [reply app/models/dated_struct.rb]: nope.

File: app/services/workflow_service/pensioner_capital_disregard.rb (hunk 2 above)
Comment (stephenrichards) [reply app/services/workflow_service/pensioner_capital_disregard.rb]: 👍

File: config/thresholds/8-Apr-2019.yml (hunk 3 above)
Comment (stephenrichards) [reply config/thresholds/8-Apr-2019.yml]: 👍

**Pull request merged**